finality-aleph/src/justification/compatibility.rs

```diff
@@ -0,0 +1,44 @@
+use crate::{
+    crypto::{Signature, SignatureV1},
+    justification::AlephJustification,
+};
+use aleph_bft::{PartialMultisignature, SignatureSet};
+use codec::{Decode, DecodeAll, Encode};
+
+/// Old format of justifications, needed for backwards compatibility.
+#[derive(Clone, Encode, Decode, Debug, PartialEq)]
+pub struct AlephJustificationV1 {
+    pub signature: SignatureSet<SignatureV1>,
+}
+
+impl From<AlephJustificationV1> for AlephJustification {
+    fn from(just_v1: AlephJustificationV1) -> AlephJustification {
+        let size = just_v1.signature.size();
+        let just_drop_id: SignatureSet<Signature> = just_v1
+            .signature
+            .into_iter()
+            .fold(SignatureSet::with_size(size), |sig_set, (id, sgn)| {
+                sig_set.add_signature(&sgn.into(), id)
+            });
+        AlephJustification {
+            signature: just_drop_id,
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum JustificationDecoding {
+    V1(AlephJustificationV1),
+    V2(AlephJustification),
+    Err,
```

Comment:
could we make the Err variant a bit more descriptive (UnknownJustificationFormatError or something to that effect?)

Reply:
I don't really know what to put there, it can really be either unknown format or error while decoding. The name `JustificationDecoding::Err` is enough descriptive in my opinion